src/copilot/tools/toolBase.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import { TelemetryViews, TelemetryActions, ActivityStatus } from "../../sharedInterfaces/telemetry";
+import { startActivity } from "../../telemetry/telemetry";
+
+export abstract class ToolBase<T> implements vscode.LanguageModelTool<T> {
+    toolName: string;
+
+    async invoke(
+        options: vscode.LanguageModelToolInvocationOptions<T>,
+        _token: vscode.CancellationToken,
+    ) {
+        const telemetryActivity = startActivity(
```

Comment:
There are already tool invocations happening before these were added, right?  Do we also capture telemetry for those tool calls?